src/go/rdctl/pkg/reg/reg.go

```diff
@@ -29,37 +29,60 @@ func escape(s string) string {
 //
 // Params:
 // pathParts: represents the registry path to the current item
-// v: the reflected value of the current field
+// structType: a reflection on the ServerSettingsForJSON struct, so we can determine how to interpret each field in the instance
+// value: the reflected value of the current field, based on a simple map[string]interface{} JSON-parse
 // jsonTag: the name of the field, used in json (and the registry)
+// path: a dotted representation of the fully-qualified name of the field
 //
 // Returns two values:
 //
 //	an array of lines representing the current value
 //	an error: the only non-nil error this function can return is when it encounters an unhandled value type
-func convertToRegFormat(pathParts []string, v reflect.Value, jsonTag string) ([]string, error) {
-	kind := v.Kind()
+func convertToRegFormat(pathParts []string, structType reflect.Type, value reflect.Value, jsonTag, path string) ([]string, error) {
+	kind := structType.Kind()
+	if value.Kind() == reflect.Interface && kind != reflect.Interface {
+		if value.IsNil() {
+			return nil, nil
+		}
+		return convertToRegFormat(pathParts, structType, value.Elem(), jsonTag, path)
+	}
+	if value.Kind() == reflect.Ptr {
+		return nil, fmt.Errorf("reg-file generation: got an unexpected pointer for %s value %v, expecting type %v", path, value, structType)
+	}
 	switch kind {
 	case reflect.Struct:
-		numFields := v.NumField()
-		scalarReturnedLines := make([]string, 0, numFields)
+		// This is very different from plist-processing of structs.
+		// In the plist world we want to preserve order of fields in the original JSON,
+		// because that's what the plutil converter does.
```

Comment:
This isn't actually true — the plist converter prefers the ordering in the type definition.  But I guess this is still different in that we're sorting (instead of using the struct order as-is), so a comment is still required (just saying different things).

Reply:
Yes, now they're more structurally similar. The sorting before walking the list is different, and processing the results is different, but the loop for `structType.Kind() == reflect.Struct` is now the same.